### PR TITLE
CFIN-445 Add sentence case function to capitalise Undergraduate tag

### PR DESCRIPTION
**Pull request by ChristianBeddows**

### CODE REVIEW

File: src/components/CourseDetails.js [hunk 1]
```diff
@@ -2,12 +2,13 @@ import React from "react";
...
 import { RichTagList } from "./RichTagList";
 import { RichTag } from "./RichTag";
+import { sentenceCase } from "../utils/formatting";
 import PropTypes from "prop-types";
 
 const CourseDetails = ({ course }) => {
     return (
         <RichTagList>
-            <LevelAndAwardTag level={course.level} award={course.award} />
+            <LevelAndAwardTag level={sentenceCase(course.level)} award={course.award} />
```
Comment (ChristianBeddows) [inline src/components/CourseDetails.js:11]: Do you think this is the right place to use this function?
I am unsure whether it belongs within the `LevelAndAwardTag` function, similar to how the startYear string manipulation happens within the `StartDateAndLengthTag` function on line 36.

Comment (mrgibbonsuoy) [reply src/components/CourseDetails.js:11]: I'd agree with your position it keeps it consistent.

**Review by mrgibbonsuoy — APPROVED** [review]

File: src/components/CourseDetails.js (hunk 1 above)
Comment (ChristianBeddows) [reply src/components/CourseDetails.js:11]: I'm not sure if this is consistent - the function call should perhaps belong on line 23, similar to line 36, further down in this file

Comment (ncn1) [reply src/components/CourseDetails.js:11]: Surely it's actually inconsistent - this one done in the _call to_ LevelAndAwardTag
 and startYear string manipulation happens _within_ the StartDateAndLengthTag.
 
 If we are saying the level description in a rich tag should always be in sentence case (which I think we are and was the implication of using css for it), I think it should also be used _within_ LevelAndAwardTag.

File: src/utils/formatting.js [hunk 2]
```diff
@@ -0,0 +1,10 @@
+const sentenceCase = (string) => {
```
Comment (ncn1) [inline src/utils/formatting.js:1]: Nice succinct and clear function!

File: src/components/CourseDetails.js (hunk 1 above)
Comment (mrgibbonsuoy) [reply src/components/CourseDetails.js:11]: I was thinking about someone opening the project and scanning the files. Whilst out of position you can see at a glance the tags.

Comment (ce666-york) [reply src/components/CourseDetails.js:11]: Given it's a `LevelAndAwardTag` rather than a generic `Tag` component, I'd lean towards putting all functionality for rendering this tag correctly within the component. It's really not a problem though.

File: src/tests/utils/formatting.test.js [hunk 3]
```diff
@@ -0,0 +1,7 @@
+import { sentenceCase } from "../../utils/formatting";
+
+describe("sentenceCase", () => {
+    it("Capitalises the first letter of the input string", async () => {
```
Comment (ce666-york) [inline src/tests/utils/formatting.test.js:4]: I'd like to see more tests for edge cases e.g. empty string, null string.

I think it would also be useful to put in tests to show what we expect to see when we receive input that's already in sentence case, input that's completely capitalised, and input with spaces/formatting. Pretend you don't know anything about the implementation details!

**Review by ce666-york — CHANGES_REQUESTED** [review]
Small point about more tests!

File: src/tests/utils/formatting.test.js (hunk 3 above)
Comment (FergusMcGlynn) [reply src/tests/utils/formatting.test.js:4]: This could probably be [shortened using Jest's `each` function](https://elfi-y.medium.com/reduce-boilerplate-test-code-with-jest-it-each-30a0eec9776d) - like the `where` clause in Spock tests. But it's not a problem how it is currently.

**Review by FergusMcGlynn — APPROVED** [review]

File: src/components/CourseDetails.js (hunk 1 above)
Comment (FergusMcGlynn) [reply src/components/CourseDetails.js]: I'm a little confused - is there any reason not to call `sentenceCase()` in `RichTag`, so that it applies to all text in all tags no matter which type? E.g, in `RichTag.js`:
```
    <li className="rich-tag">                                                                                     
        {topIcon && <i aria-hidden className={`c-icon c-icon--${topIcon} c-icon--2x`} data-testid="tag-icon" />}  
        <p className="rich-tag__title">{sentenceCase(title)}</p>                                                                
        <p className="rich-tag__text">{sentenceCase(subText)}</p>                                                               
    </li>                                                                                                         
```

**Review by ce666-york — APPROVED** [review]

File: src/components/CourseDetails.js (hunk 1 above)
Comment (ChristianBeddows) [reply src/components/CourseDetails.js]: I like the idea of using the function within `RichTag` so that it always applies to text - though following from our discussion earlier I'm going to leave it as it is, in the idea of changing as little as possible within this changeset. I think it could be something to revisit though!
